server/controllers/ArticleController.js

```diff
@@ -0,0 +1,201 @@
+          .createArticleResponse(article, username, bio, image, slug),
+        message: 'Article Sucessfully Created',
+      })).catch(next);
+  }
+
+  /**
+   * Method gets an article from the database,
+   * It returns a json response of the article retrieved
+   * It handles the GET /api/articles/:slug.
+   * @param {Object} req the request body
+   * @param {Object} res the response body
+   * @param {function} next a call to the next function
+   * @returns {Object} the response body
+   */
+  static getArticle(req, res, next) {
+    Article.find({
+      where: { slug: req.params.slug },
+      include: [{
+        model: User,
+        as: 'author',
+        attributes: [
+          'username', 'bio', 'image',
+        ],
+      }],
+      attributes: [
+        'slug', 'title', 'description',
+        'body', 'createdAt', 'updatedAt',
+      ],
+    })
+      .then((article) => {
+        if (!article) {
+          return res.status(404).json({
+            status: 'fail',
+            message: 'Article not found',
+          });
+        }
+        return res.status(200).json({
+          article,
+          message: 'Article Found',
+        });
+      })
+      .catch(next);
+  }
+
+  /**
+   * Method gets all articles from the database,
+   * It returns a json response of the article retrieved
+   * It handles the GET /api/articles.
+   * @param {Object} req the request body
+   * @param {Object} res the response body
+   * @param {function} next a call to the next function
+   * @returns {Object} the response body
+   */
+  static getAllArticle(req, res, next) {
+    Article.findAndCountAll({
+      include: [{
+        model: User,
+        as: 'author',
+        attributes: [
+          'username', 'bio', 'image',
+        ],
+      }],
+      attributes: [
+        'slug', 'title', 'description',
+        'body', 'createdAt', 'updatedAt',
+      ],
+      order: [['createdAt', 'DESC']],
+    })
+      .then((articles) => {
+        if (!articles.count) {
+          return res.status(404).json({
+            status: 'fail',
+            message: 'No Articles found',
+          });
+        }
+        return res.status(200).json({
+          articles: articles.rows,
+          articlesCount: articles.count,
+        });
+      })
+      .catch(next);
+  }
+
+  /**
+   * Method updates an article in the database,
+   * It returns a json response of the article updated
+   * It handles the PUT /api/articles/:slug.
+   * @param {Object} req the request body
+   * @param {Object} res the response body
+   * @param {function} next a call to the next function
+   * @returns {Object} the response body
+   */
+  static updateArticle(req, res, next) {
+    const {
+      title, description, body,
+    } = req.body.article;
+    const authorId = req.body.decoded.id;
+    const fieldsArr = [];
+
+    if (title) fieldsArr.push('title');
+    if (description) fieldsArr.push('description');
+    if (body) fieldsArr.push('body');
```

Comment:
What is the essence of checking for these fields if you end up using the same values as you update parameter?

Reply:
Im checking for the fields that require updates which may be some or all of the fields and making updates accordingly